Source Main 5.2/source/ZzzInterface.cpp

```diff
@@ -6149,16 +6148,8 @@ void AttackWizard(CHARACTER* c, int Skill, float Distance)
                     }
                     to->Angle[2] = CreateAngle2D(to->Position, tc->TargetPosition);
 
-                    if (Teleport)
-                    {
-                        Teleport = false;
-                    }
-                    else
-                    {
-                        Teleport = true;
-                        SocketClient->ToGameServer()->SendTeleportTarget(tc->Key, TargetX, TargetY);
-                        SetPlayerTeleport(tc);
-                    }
```

Comment:
Removing the Teleport boolean checks here is the actual fix to the teleport bug, but I think it's better to remove the Teleport variable altogether. if ever it's needed, it's best to check o->Teleport value instead.

Reply:
Yes, global variables are a bad idea anyway 👍